charts/neon-proxy/templates/certificate.yaml

```diff
@@ -0,0 +1,17 @@
+{{- if .Values.settings.domain }}
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: {{ include "neon-proxy.certificate" . }}
+spec:
+  commonName: '{{ .Values.settings.domain }}'
```

Comment:
i prefer use `{{ .Values.settings.domain | quote }}` as you can use or not use quotes in values file. This approach more reliable. The same with other stings that may have special chars.